src/main/java/me/THEREALWWEFAN231/tunnelmc/gui/BedrockConnectionScreen.java

```diff
@@ -0,0 +1,113 @@
+package me.THEREALWWEFAN231.tunnelmc.gui;
+
+import it.unimi.dsi.fastutil.booleans.BooleanConsumer;
+import me.THEREALWWEFAN231.tunnelmc.bedrockconnection.Client;
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.client.gui.screen.ScreenTexts;
+import net.minecraft.client.gui.widget.ButtonWidget;
+import net.minecraft.client.gui.widget.TextFieldWidget;
+import net.minecraft.client.util.math.MatrixStack;
+import net.minecraft.text.LiteralText;
+import net.minecraft.text.TranslatableText;
+
+@Environment(EnvType.CLIENT)
+public class BedrockConnectionScreen extends Screen {
+	
+//	The UI that comes up after you click on Connect To Bedrock.
+	
+	private ButtonWidget selectServerButton;
+	private TextFieldWidget addressField;
+	private TextFieldWidget portField;
+	private BooleanConsumer callback;
+	private final Screen parent;
+
+	public BedrockConnectionScreen(Screen parent, BooleanConsumer callback) {
+		super(new LiteralText("Bedrock Connection"));
+		this.parent = parent;
+		this.callback = callback;
+	}
+
+	public void tick() {
+		this.addressField.tick();
+		this.portField.tick();
+	}
+
+	public boolean keyPressed(int keyCode, int scanCode, int modifiers) {
+		if ((this.getFocused() != this.addressField && this.getFocused() != this.portField)
+				|| keyCode != 257 && keyCode != 335) {
```

Comment:
Is it possible to change these to enum values or something not hardcoded?

Reply:
Not sure, this is the code used in the DirectConnection.class file.

Reply:
Yes, I was going to change that later.